### PR TITLE
ADBDEV-4986: Implemented service to reload profile

### PR DESCRIPTION
**Pull request by RomaZe**

### CODE REVIEW

File: server/pxf-service/src/main/java/org/greenplum/pxf/service/profile/ProfileReloadServiceImpl.java [hunk 1]
```diff
@@ -1,12 +1,91 @@
...
+            log.info("Skipping reloading profile '{}' for server '{}'", profile, server);
+        }
+    }
+
+    private Plugin getAccessor(String profile) {
+        try {
+            Map<String, String> pluginMap = profileConf.getPlugins(profile);
+            return Optional.ofNullable(pluginMap)
+                    .map(plugins -> getAccessorInstance(plugins.get(ACCESSOR_KEY)))
+                    .orElse(null);
```
Comment (SidorovGreg) [inline server/pxf-service/src/main/java/org/greenplum/pxf/service/profile/ProfileReloadServiceImpl.java]: Better to return `Optional<Plugin>`

Comment (RomaZe) [reply server/pxf-service/src/main/java/org/greenplum/pxf/service/profile/ProfileReloadServiceImpl.java]: Replaced null by Optional

File: server/pxf-service/src/main/java/org/greenplum/pxf/service/profile/ProfileReloadServiceImpl.java [hunk 2]
```diff
@@ -1,12 +1,89 @@
...
+    }
+
+    private Plugin getAccessorInstance(String accessorClassName) {
+        try {
+            Plugin instance = basePluginFactory.getPluginInstance(accessorClassName);
+            log.debug("Initialize instance {} of the accessor class {}", instance, accessorClassName);
+            return instance;
+        } catch (Exception e) {
+            log.error("Failed to initialize instance of the accessor class {}. {}", accessorClassName, e.getMessage());
+            return null;
```
Comment (iamlapa) [inline server/pxf-service/src/main/java/org/greenplum/pxf/service/profile/ProfileReloadServiceImpl.java]: Why was this exception not propagated further? How will the user know that the reload was not entirely successful?

Comment (RomaZe) [reply server/pxf-service/src/main/java/org/greenplum/pxf/service/profile/ProfileReloadServiceImpl.java]: Some of the custom profiles might not contain accessor class defined in the `pxf-profile.xml`. For example,
```
<?xml version="1.0" encoding="UTF-8"?>
<profiles>
    <profile>
        <name>test:csv</name>
        <plugins>
            <metadata>Y</metadata>
        </plugins>
    </profile>
</profiles>
``` 
So, we cannot initialized accessor class for this profile. But we want to reload other profiles in the list.

**Review by SidorovGreg — APPROVED** [review]

**Pull request merged**